ISISComputingGroup/EPICS-ioc
# Ticket2787: Harmonise Galil Motor ioc

### Description of work

See https://github.com/ISISComputingGroup/IBEX/issues/2787

### To test

* Ensure you have pulled and built all three PRs
* Run the associated upgrade script
* Ensure that:
    * When adding a galil to a configuration in the GUI it prompts for the new style macros
    * The generic axis.cmd is loaded for each galil axis 
    * Galil parameters are autosaved and recovered correctly
    * Galil axes still contribute to the motors moving display and are stopped by stop all
    * The position of a galil in the Table of Motors can be set by MTRCTRL
    * The IOC still works for a simulated motor and a real controller

Notes: 
* Loading individual axes at runtime was not done as it would require even larger changes to configurations and to underlying galil support dbs
* RECSIM was not added due to https://github.com/ISISComputingGroup/IBEX/issues/2736

Confirm documentation at 
https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Motor-IOCs

---

#### Code Review

    - [PV naming](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/PV-Naming).
    - [Disable records](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Disable-records)
    - [Record simulation](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Record-Simulation)
    - [Finishing touches](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/IOC-Finishing-Touches)

### Functional Tests

- IOC responds correctly in:

### Final steps
